app/views/carts/_cart.html.erb

```diff
@@ -17,4 +17,6 @@
 <%= button_to 'Empty cart', cart, method: :delete,
     data: { confirm: 'Are you sure?' } %>
 <br/>
-<%= link_to 'Back', '/' %>
+<% if request.path == "/orders/new" %>
+  <%= link_to 'Back', '/' %>
```

Comment:
You should consider using the route names rather than absolute paths. What happens if you change your routes file?